### PR TITLE
#28: Automatic Google Analytics properties for Events (closes #28)

### LINKED ISSUE #28: Automatic Google Analytics properties for Events
[Trello card](https://trello.com/c/58468f8200143303547412dc/)

#specs

* If `utm_` values are present in the querystring, add them to the ContactHub cookie.
* set expiry time at 1 year for ContactHub cookie.
* whenever an `Event` is sent to ContactHub, add a `GoogleAnalytics` property containing the `utm_` values.

### PR DESCRIPTION
**Pull request by dhinus**
Closes #28

### tests performed
* ✅ `npm test`
* ✅ `npm run build`
* Tested manually with `npm run example`: utm_ params were stored in the cookie and sent to the API

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

File: src/contacthub.js [hunk 1]
```diff
@@ -23,6 +23,11 @@ const varName: string = window.ContactHubObject || 'ch';
 const cookieName: string = window.ContactHubCookie || '_ch';
 const apiUrl: string = window.ContactHubAPI || 'https://api.contactlab.it/hub/v1';
 
+function getQueryParam(name) {
+  const match = RegExp(`[?&]${name}=([^&]*)`).exec(window.location.search);
+  return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
+}
```
Comment (ascariandrea) [inline src/contacthub.js]: 😎

File: src/contacthub.js [hunk 2]
```diff
@@ -32,32 +37,42 @@ const getCookie = (): ContactHubCookie => {
...
   const _ch = cookies.getJSON(cookieName) || {};
 
+  // read Google Analytics query params if present
+  const utm_source = getQueryParam('utm_source');
+
+  if (utm_source) {
+    // Store ga values in the ch cookie, overwriting any previous ga value.
+    _ch.ga = {
+      utm_source,
+      utm_medium: getQueryParam('utm_medium') || undefined,
```
Comment (ascariandrea) [inline src/contacthub.js]: The fallback to `undefined` could be inserted in `getQueryParam` function

File: test/ga.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      requests.push(xhr);
+    };
+  });
+
+  const setConfig = () => {
+    _ch('config', config);
+  };
+
+  it('stores utm_* vars in the ch cookie', () => {
+    // FIXME: find a good way to mock window.location.search
```
Comment (ascariandrea) [inline test/ga.js]: Are you going to fix this on another pr?

File: src/contacthub.js (hunk 2 above)
Comment (dhinus) [inline src/contacthub.js]: good point.

File: test/ga.js (hunk 3 above)
Comment (dhinus) [inline test/ga.js]: yes, that was the idea.

File: src/contacthub.js (hunk 2 above)
Comment (dhinus) [inline src/contacthub.js]: Done.

**Pull request merged**